tests/test_hypothesis.py

```diff
@@ -0,0 +1,186 @@
+import warnings
+
+import pandas as pd
+import pytest
+from unittest.mock import patch
+from comma.hypothesis import Hypothesis
+
+
+class TestHypothesis:
+    @pytest.fixture
+    def setup_time_period(self):
+        data = {
+            "starting_date": "2020-01-01",
+            "steps": 5,
+            "date_format": "%Y-%m-%d",
+            "expected_result": ("2020-01-01", "2020-01-06"),
+        }
+        return data
+
+    @pytest.fixture
+    def mock_df(self):
```

Comment:
I think the name is slightly misleading because nothing is mocked here.

```suggestion
    def dummy_df(self):
```